lib/Doctrine/DBAL/Types/Type.php

```diff
@@ -201,6 +215,11 @@ public static function getType($name)
      */
     public static function addType($name, $className)
     {
+        self::warnAgainstUsingTheOldAPI();
+        if ($name === $className) {
+            TypeRegistry::addType($className);
```

Comment:
@Majkl578 that kind of thing can't be done if `TypeRegistry` is not static, unless maybe we add a private static variable to hold the registry?

Reply:
Yes, in 2.x it'd have to be done that way.
Then in 3.0, it should probably be configured via Configuration object and use TypeRegistry instance instead of static calls to Type across DBAL/ORM.

Reply:
Ok will change to non-static.